xarray in blending code

mats-knmi opened the pull request:
This PR implements xarray into the blending code and fixes the associated tests

mats-knmi committed:
fix blend utils and xarray dataformat

mats-knmi committed:
fix test blending utils

mats-knmi committed:
fix steps blending tests

mats-knmi committed:
Merge remote-tracking branch 'origin/xarray/main' into xarray/blend-utils

larsbonnefoy reviewed (commented):
Looks fine for me, many changes at once make it a bit difficult to follow along. 
I've seen that you touched the "transform" values in the meta data.
Have you moved that information elsewhere ?

mats-knmi commented:
> Looks fine for me, many changes at once make it a bit difficult to follow along. I've seen that you touched the "transform" values in the meta data. Have you moved that information elsewhere ?

Netcdf doesn't allow you to write attributes with None values, so I removed the transform attribute when it was set to None. The checks for this attribute are now changed to `if "transfom" is in attrs` in stead of `if attrs["transform"] is None`. So functionally nothing changed.

sidekock commented:
Is this pull request still active or do we have a more updated one with the one I just commented on?

mats-knmi commented:
@sidekock The PR you just commented on contains all the changes in this PR and more. I tried to split up these changes in 2 PRs, but the other pull request extends on the changes in this one, so that contains all these changes plus the extra changes. Once we merge this PR, the other one will be a lot clearer since then that one will only contain the extra changes.

But since you reviewed the other one already, you don't have to also look at  this one.
